ios/OwrHelpers/OpenWebRTCNativeHandler.m

```diff
@@ -0,0 +1,836 @@
+//  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+//  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+//  OF SUCH DAMAGE.
+//
+
+#import "OpenWebRTCNativeHandler.h"
+
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include "owr.h"
+#include "owr_audio_payload.h"
+#include "owr_audio_renderer.h"
+//#include "owr_local.h"
+#include "owr_media_session.h"
+//#include "owr_transport_agent.h"
+#include "owr_video_payload.h"
+#include "owr_video_renderer.h"
+#include "owr_window_registry.h"
+
+#include <gio/gio.h>
+//#include <json-glib/json-glib.h>
+//#include <libsoup/soup.h>
+#include <string.h>
+
+#define SELF_VIEW_TAG "self-view"
+#define REMOTE_VIEW_TAG "remote-view"
+
+OwrVideoRenderer *renderer;
+
+static OpenWebRTCNativeHandler *staticSelf;
+
+@implementation OpenWebRTCNativeHandler
+
+- (instancetype)init
+{
+    if (self = [super init]) {
+        owr_init();
+        staticSelf = self;
+    }
+    return self;
+}
+
+- (void)setSelfView:(OpenWebRTCVideoView *)selfView
+{
+    owr_window_registry_register(owr_window_registry_get(), SELF_VIEW_TAG, (__bridge gpointer)(selfView));
+}
+
+- (void)setRemoteView:(OpenWebRTCVideoView *)remoteView
+{
+    owr_window_registry_register(owr_window_registry_get(), REMOTE_VIEW_TAG, (__bridge gpointer)(remoteView));
+}
+
+- (void)handleOfferReceived:(NSString *)offer
+{
+
+}
+
+- (void)handleRemoteCandidateReceived:(NSString *)candidate
+{
+
+}
+
+- (void)startGetCaptureSources:(OwrMediaType)types
+{
+    owr_get_capture_sources(types, (OwrCaptureSourcesCallback)got_local_sources, NULL);
+}
+
+#pragma mark - C stuff
+
+static GList *local_sources, *renderers;
+static OwrTransportAgent *transport_agent;
+//static gchar *session_id, *peer_id;
+static gchar *peer_id;
+//static guint client_id;
+static gchar *candidate_types[] = { "host", "srflx", "relay", NULL };
+static gchar *tcp_types[] = { "", "active", "passive", "so", NULL };
+
+static void read_eventstream_line(GDataInputStream *input_stream, gpointer user_data);
+static void got_local_sources(GList *sources);
+
+static void got_remote_source(OwrMediaSession *media_session, OwrMediaSource *source,
+                              gpointer user_data)
+{
+    OwrMediaType media_type;
+    gchar *name = NULL;
+    OwrMediaRenderer *renderer;
+
+    g_return_if_fail(OWR_IS_MEDIA_SESSION(media_session));
+    g_return_if_fail(!user_data);
+
+    g_object_get(source, "media-type", &media_type, "name", &name, NULL);
+    g_message("Got remote source: %s", name);
+
+    if (media_type == OWR_MEDIA_TYPE_AUDIO)
+        renderer = OWR_MEDIA_RENDERER(owr_audio_renderer_new());
+    else if (media_type == OWR_MEDIA_TYPE_VIDEO)
+        renderer = OWR_MEDIA_RENDERER(owr_video_renderer_new(NULL));
```

Comment:
Do you really mean NULL here and not a specific tag? This would cause glimagesink to create its own window...? Or try to and probably fail on iOS.